console-subscriber/src/lib.rs

```diff
@@ -922,7 +922,39 @@ impl Server {
     /// ```
     /// [`serve_with`]: Server::serve_with
     pub async fn serve(self) -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
-        self.serve_with(tonic::transport::Server::default()).await
+        self.serve_with(tonic::transport::Server::builder()).await
+    }
+
+    #[cfg(feature = "grpc-web")]
+    /// Starts the gRPC service with the default gRPC settings and gRPC-Web
+    /// support.
+    pub async fn serve_with_grpc_web(
```

Comment:
I am not sure if there is a better way to do this. I copied the `start_with` function here.

Reply:
Looking at this function, it seems that it's exactly the same as `serve_with()` except for the `.accept_http1(true)` line. Is that correct?

If that's the case, I'm not even sure that we need a separate method for this. Instead, could the user of the API do the following:

```rust
let (console_layer, server) = console_subscriber::ConsoleLayer::new();
// Do something with the console_layer
server.serve_with(tonic::transport::Server::default().accept_http1(true)).await;
```
In that case, I think we can add documentation to `serve_with` to that effect.

I still think that perhaps being able to use `spawn` to launch a fully configured grpc-web server is valuable (otherwise, gprc-web forces the user to use the super custom way to define everything, which has some drawbacks).

What do you think?

Reply:
> Looking at this function, it seems that it's exactly the same as `serve_with()` except for the `.accept_http1(true)` line. Is that correct?

It's not only about HTTP 1.1, also we need to use `tonic_web::enable(instrument_server)` to wrap the service.

Reply:
It looks like maybe it's possible to do this by adding a layer: `.layer(GrpcWebLayer::new())`?

See the second example here: https://docs.rs/tonic-web/latest/tonic_web/

Reply:
After we added the layer, it would not be a server and we can not pass it to serve_with.

Reply:
It looks like it's still a server, but not an identity server (`Server<L = [Identity]>`). I wonder if we can get away with modifying `serve_with` to accept a generic server without making it a breaking API change.

Reply:
Although we can pass it to `serve_with()`, I don't think it's a good API for users. This is because `GrpcWebLayer::new()` doesn't set the CORS configuration, which is usually required for web applications.

Reply:
@hds Could you please provide some help here? I am not sure what kind of generic I can use here. Because we want to accept both `Server<L = [Identity]>` and `Server<
            tower::layer::util::Stack<
                tonic_web::GrpcWebLayer,
                tower::layer::util::Stack<
                    tower_http::cors::CorsLayer,
                    tower::layer::util::Identity,
                >,
            >,
        >` here. Do you have any thoughts or suggestion?